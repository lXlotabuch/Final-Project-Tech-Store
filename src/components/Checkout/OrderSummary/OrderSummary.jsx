import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { selectCartSummary, selectShippingCost } from '../../../store/cart/reducer';
import StyledOrderSummary from './StyledOrderSummary';
import { RowBetween } from '../../CartPage/Flex';

export const OrderSummaryComponent = ({summary, shippingCost}) => (
  <StyledOrderSummary>
    <h4>Order summary</h4>
    <RowBetween>
      <div>
        <p className="sumtotal">
          Items total:
        </p>
      </div>
      <div>
        <span className="sumtotal">
          ₴
          {summary}
        </span>
      </div>
    </RowBetween>
    <RowBetween>
      <div>
        <p className="sumtotal">
          Shipping:
        </p>
      </div>
      <div>
        <span className="sumtotal">
          ₴
          {shippingCost}
        </span>
      </div>
    </RowBetween>
    <div className="line">
      <RowBetween>
        <div>
          <p className="ordertotal">
            Order total:
          </p>
        </div>
        <div>
          <span className="ordertotal">
            {summary + shippingCost}
            ₴
          </span>
        </div>
      </RowBetween>
    </div>
  </StyledOrderSummary>
)

const mapStateToProps = (state) => ({
  summary: selectCartSummary(state),
  shippingCost: selectShippingCost(state),
})

const OrderSummary = connect(mapStateToProps, null)(OrderSummaryComponent)

OrderSummaryComponent.propTypes = {
  summary: PropTypes.number.isRequired,
  shippingCost: PropTypes.number.isRequired,
}

export default OrderSummary;