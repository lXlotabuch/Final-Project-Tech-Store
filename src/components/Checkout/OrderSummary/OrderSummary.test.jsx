import React from 'react';
import { render } from '@testing-library/react';
import { OrderSummaryComponent } from './OrderSummary';

describe('OrderSummary', () => {
  test('render', () => {
    const {asFragment} = render(<OrderSummaryComponent
      summary={125478}
      shippingCost={124}
    />)

    expect(asFragment()).toMatchSnapshot()
  })
})
