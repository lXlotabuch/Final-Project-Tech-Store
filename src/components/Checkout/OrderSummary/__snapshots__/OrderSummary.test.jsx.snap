// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSummary render 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ sc-gsTCUz fnKmPD eTbcwS"
  >
    <h4>
      Order summary
    </h4>
    <div
      class="sc-dlfnbm sc-gKsewC jcczqT kcTMhS"
    >
      <div>
        <p
          class="sumtotal"
        >
          Items total:
        </p>
      </div>
      <div>
        <span
          class="sumtotal"
        >
          ₴125478
        </span>
      </div>
    </div>
    <div
      class="sc-dlfnbm sc-gKsewC jcczqT kcTMhS"
    >
      <div>
        <p
          class="sumtotal"
        >
          Shipping:
        </p>
      </div>
      <div>
        <span
          class="sumtotal"
        >
          ₴124
        </span>
      </div>
    </div>
    <div
      class="line"
    >
      <div
        class="sc-dlfnbm sc-gKsewC jcczqT kcTMhS"
      >
        <div>
          <p
            class="ordertotal"
          >
            Order total:
          </p>
        </div>
        <div>
          <span
            class="ordertotal"
          >
            125602₴
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
